hooks/pre-checkout

```diff
@@ -19,10 +19,14 @@ DESTINATION=/tmp/$BUILDKITE_PIPELINE_SLUG.git.tar
 # If we have an available Git Mirror Server, use it
 if [ -n "${GIT_MIRROR_SERVER_ROOT:-}" ]; then
 	echo "Using Git Mirror Server at $GIT_MIRROR_SERVER_ROOT"
-	URL=$(curl "$GIT_MIRROR_SERVER_ROOT/manifest" | grep $REPO_PATH)
+	if URL=$(curl "$GIT_MIRROR_SERVER_ROOT/manifest" | grep $REPO_PATH$); then
```

Comment:
This could fail for multiple reasons (maybe the URL isn't accessible?)

IMHO if we're doing it this way we should download the file to some temp directory (possible doing error handling on the download process), _then_ do the `grep` for the repo path do that error handling.

Reply:
Also, IINM we may want to do ` | grep "$REPO_PATH$"`, but idk if that actually works? Just thinking if `$REPO_PATH` contains a space this would break. `shellcheck` could probably help lint this for us

Reply:
a bit unclear here as I had used `$REPO_PATH$` in the above already?